### PR TITLE
GOVUKAPP-515 Add Lottie and Launch Screen animation

### PR DESCRIPTION
**Pull request by batesyuk**
Added animation to splash screen.

Updated snapshot test case names (because I got it wrong initially)

There should be no further work required to support testing animations assuming you use the new `AnimationView` class to load them.

`AnimationView` is both to aid snapshot testing, but also will allow us to switch out animations for static images etc if required, it has the other benefit of siloing Lottie to a single class should we decide to replace it.

Reduced line length in SwiftLint to 100. It is achievable, and we currently only have one line that exceeds that (fixed), and it's much nicer for PRs when lines are below 100. Should this be an issue, its easier to raise it rather than lower it later.

### CODE REVIEW

File: Production/govuk_ios/Views/AnimationView.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func configureUI() {
+        internalAnimationView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(internalAnimationView)
+    }
+
+    private func configureConstraints() {
+        internalAnimationView.topAnchor.constraint(equalTo: topAnchor).isActive = true
```
Comment (raphaelolu) [inline Production/govuk_ios/Views/AnimationView.swift:33]: minor, but perhaps would be best to wrap these in a  NSLayoutConstraint.activate([]), may save you abit of code without having to write isActive = true

**Review by raphaelolu — CHANGES_REQUESTED** [review]

File: .swiftlint.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ trailing_whitespace:
 vertical_whitespace:
   max_empty_lines: 2
 line_length:
-  warning: 120
+  warning: 100
```
[the remark this answers is not in the document]
Comment (batesyuk) [reply .swiftlint.yml:18]: Yep, as per PR description

[the remark this answers is not in the document]
Comment (batesyuk) [reply .swiftlint.yml:18]: Open for discussion though

File: Production/govuk_ios/ViewControllers/LaunchViewController.swift [hunk 3]
```diff
@@ -2,24 +2,50 @@ import UIKit
...
-        activityIndicator.centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
-        activityIndicator.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        animationView.translatesAutoresizingMaskIntoConstraints = false
+        animationView.rightAnchor.constraint(
+            equalTo: view.layoutMarginsGuide.rightAnchor,
+            constant: -20
+        ).isActive = true
+        animationView.leftAnchor.constraint(
+            equalTo: view.layoutMarginsGuide.leftAnchor,
+            constant: 20
```
Comment (robinphillips) [inline Production/govuk_ios/ViewControllers/LaunchViewController.swift:46]: Should these two constant values of `-20` and `20` both use the same constant set somewhere in the class? `animationInset` or something?

File: Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift [hunk 4]
```diff
@@ -1,13 +1,18 @@
...
 @testable import govuk_ios
 
 class SnapshotTestCase: FBSnapshotTestCase {
 
     override func setUp() {
         super.setUp()
+        Container.shared.lottieConfiguration.register {
+            LottieConfiguration(renderingEngine: .mainThread)
+        }
 //        self.recordMode = true
```
Comment (robinphillips) [inline Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift:16]: Commented code

File: .swiftlint.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (robinphillips) [reply .swiftlint.yml:18]: It's easy enough to change it back if it becomes a problem later.

File: Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift (hunk 4 above)
Comment (batesyuk) [reply Tests/govuk_ios/govuk_ios_snapshot_tests/Helpers/SnapshotTestCase.swift:16]: This isn't part of this PR. It was added as part of the introduction of Snapshot testing.
It is helpful if you need to retake all snapshot references for some reason (new iPhone). We can take it out and add some documentation to the README about it if preferred, but I will do it in another PR

File: Production/govuk_ios/ViewControllers/LaunchViewController.swift (hunk 3 above)
Comment (batesyuk) [reply Production/govuk_ios/ViewControllers/LaunchViewController.swift]: I have added layout margins to the BaseViewController instead

**Review by 1pretz1 — APPROVED** [review]
LGTM - nice work!

**Review by raphaelolu — APPROVED** [review]

**Pull request merged**